### PR TITLE
Missing docs

### PR DESCRIPTION
**Pull request by gciltaru**
#### Purpose
- Add missing docs
- Handle multipart scenarios
- Select the specific scenario number for scenarios with same criteria

### CODE REVIEW

**Comment by remear** [comment]
Could you add more context here for what problem the scenario number aspect aims to solve.

We specifically didn't bother with the multipart stuff because we didn't want to create a new extension package every time we run this suite. I recall that mainly being due to our inability to delete them afterward, which creates cruft in the DB (also requires the version bumping each time), EFS, and Akamai. While this is fine when we run it against previews because they're isolated and ephemeral, doing so against any static environment is undesirable. I'm interested in hearing others' opinions on what's ideal here.

**Comment by gciltaru** [comment]
> against

The scenario number helps with the `discontinue_an_extension_package` scenario. The api specification file has 2 scenarios that match the scenario name, endpoint and method: one for the setup (private release) and one for the actual step. We want a way to specify that we need the second scenario and the scenario number seemed a simple way to accomplish this.

We are creating multiple extension packages currently. Most of them are deleted, but some are not (for the private release scenario). All of these extension packages are simple and do not have any actions, events, etc.... The argument for doing this is having the the extension attributes in sync always and not manage it by hand.

**Comment by spencerroan** [comment]
For Akamai we basically have unlimited storage, and this will not affect download volume where the cost lies. This is less burdensome than a build which we allow freely. It should be pretty small on EFS, but we can clean that up. Left unchecked EFS could become an issue. We can also run this locally rather than against a static env. We could tag these tests in a way to ensure they only run on ephemeral envs and find a way to stitch that output into the normal output.

I think we should document the api with actual requests/responses when the cost burden (and other burdens) is low. When there are side effects in other systems we either need to be able to clean those up, or reduce the frequency at which we run those tests. In the case of Private EP, it should not be a huge problem for the frequency at which we regenerate documentation. There are a number of ways to solve this. 1: Cleanup afterward. such as add a hidden attribute on EP, a worker looks for that attribute and fully cleans up DB/EFS/Akamai after the fact, or give the EP a special name and allow list EP names to clean up after. Add a manual rake task to cleanup and run it afterwards. 2. Run these tests(that leave long lived artifacts on other systems) less frequently and find a way to separate them in the api spec so parts of it can stick around. This is more of a mitigation. 3. Ensure through code that these tests are only run on ephemeral envs.

We currently run this suite a few times a month, which is nothing compared to a synthetic or user usage of tables like builds revisable resources.

File: _plugins/docs.rb [hunk 1]
```diff
@@ -96,33 +97,24 @@ def form(form_name)
...
       r['endpoint'] == endpoint['path'] &&
-        r['method'] == method
+        r['method'] == method &&
+        r['documentation'] == true
     end
   end
 
   def scenario_select_by_name(name)
-    @spec['scenarios'].find { |s| s['name'] == name }
+    @spec['scenarios'].find { |s| s['name'] == name && s['documentation'] == true }
```
Comment (spencerroan) [inline _plugins/docs.rb]: this is certainly more straight forward.

File: api/reference/1.0/app_configurations/relationships/app_surfaces/app_configuration_list_relationship.md [hunk 2]
```diff
@@ -6,4 +6,4 @@ title: List relationships
 
 Retrieve the relationship linkage for the AppSurfaces related to the specified AppConfiguration.
 
-{% scenario app_configurations_relationships_app_surfaces.show app_configurations.list_the_app_surfaces_for_an_app_configuration %}
+{% scenario app_configurations_relationships_app_surfaces.show app_configurations.list_the_app_configuration_relationships_for_an_app_configuration %}
```
Comment (spencerroan) [inline api/reference/1.0/app_configurations/relationships/app_surfaces/app_configuration_list_relationship.md]: this rename is wrong

File: _plugins/docs.rb [hunk 3]
```diff
@@ -11,6 +11,7 @@
   site.pages = pages
 
   site.config['hierarchy'] = generate_hierarchy(site)
+
 end
```
Comment (spencerroan) [inline _plugins/docs.rb]: newline?

**Review by spencerroan — COMMENTED** [review]
I like that it is specific.

File: _plugins/docs.rb [hunk 4]
```diff
@@ -100,21 +100,20 @@ def form(form_name)
   def scenario(scenario_name, endpoint)
     method = endpoint&.fetch('methods')&.first
 
-    scenario_select_by_name(scenario_name) ||
-      scenario_select_by_endpoint_method(endpoint, method)
+    scenario_select_by_endpoint_method(endpoint, method) ||
+      scenario_select_by_name(scenario_name)
```
Comment (remear) [inline _plugins/docs.rb:103]: What prompted flipping the order of these?

Comment (gciltaru) [reply _plugins/docs.rb:103]: `scenario_select_by_name_endpoint_method(scenario_name, endpoint, method)` was removed.

File: _plugins/docs.rb [hunk 5]
```diff
@@ -99,20 +100,10 @@ def form(form_name)
...
 
-  def scenario_select_by_name_endpoint_method(name, endpoint, method)
-    return unless name && endpoint && method
-    @spec['scenarios'].find do |r|
-      r['name'] == name &&
-        r['endpoint'] == endpoint['path'] &&
-        r['method'] == method
-    end
-  end
-
```
Comment (spencerroan) [inline _plugins/docs.rb:115]: gonna take me a while to wrap my head around the removal of this.

Comment (spencerroan) [reply _plugins/docs.rb:115]: thanks for walking me through this. Looks good to me, though some of the file paths are a bit much, but I guess they need to be unique so jekyll does not freak out and serve the wrong stuff.

**Review by remear — DISMISSED** [review]

**Review by remear — APPROVED** [review]

**Pull request merged**